wormhole/file_transport.go

```diff
@@ -250,12 +253,30 @@ func (t *fileTransport) connectDirect(otherTransit *transitMsg) (net.Conn, error
 	return conn, nil
 }
 
-func (t *fileTransport) connectToRelay(ctx context.Context, addr string, successChan chan net.Conn, failChan chan string) {
+func (t *fileTransport) connectToRelay(ctx context.Context, successChan chan net.Conn, failChan chan string) {
 	var d net.Dialer
-	conn, err := d.DialContext(ctx, "tcp", addr)
-	if err != nil {
-		failChan <- addr
-		return
+	var conn net.Conn
+	var err error
+	addr := t.relayURL.Addr()
+
+	switch t.relayURL.Proto {
+	case "tcp":
+		conn, err = d.DialContext(ctx, "tcp", addr)
+
+		if err != nil {
+			failChan <- addr
+			return
+		}
+	case "ws", "wss", "http", "https":
```

Comment:
What's the reason to allow `http{s,}` for a websocket transport?

Reply:
I'm not absolutely sure this is necessary. My thinking was that the other end (relay in this case) determines what protocol scheme it wants to use so it would be polite to be flexible on this end to make integration with relay implementations easier.